java/flight/flight-core/src/test/java/org/apache/arrow/flight/auth2/TestBasicAuth2.java

```diff
@@ -49,117 +49,239 @@
-                                  StreamListener<FlightInfo> listener) {
-            if (!context.peerIdentity().equals(USERNAME)) {
-              listener.onError(new IllegalArgumentException("Invalid username"));
-              return;
-            }
-            listener.onCompleted();
-          }
-
-          @Override
-          public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
-            if (!context.peerIdentity().equals(USERNAME)) {
-              listener.error(new IllegalArgumentException("Invalid username"));
-              return;
-            }
-            final Schema pojoSchema = new Schema(ImmutableList.of(Field.nullable("a",
-                    Types.MinorType.BIGINT.getType())));
-            try (VectorSchemaRoot root = VectorSchemaRoot.create(pojoSchema, allocator)) {
-              listener.start(root);
-              root.allocateNew();
-              root.setRowCount(4095);
-              listener.putNext();
-              listener.completed();
-            }
-          }
-        }).headerAuthenticator(new BasicCallHeaderAuthenticator(validator)).build());
-    client = FlightClient.builder(allocator, server.getLocation()).build();
+    final BasicCallHeaderAuthenticator.AuthValidator validator =
+        new BasicAuthValidator(credentialValidator, authTokenManager);
+    final BasicCallHeaderAuthenticator basicCallHeaderAuthenticator = new BasicCallHeaderAuthenticator(validator);
+    final GeneratedBearerTokenAuthHandler generatedBearerTokenAuthHandler =
+            new GeneratedBearerTokenAuthHandler(new BasicCallHeaderAuthenticator(validator));
+    final FlightProducer flightProducer = new NoOpFlightProducer() {
+      @Override
+      public void listFlights(CallContext context, Criteria criteria,
+                              StreamListener<FlightInfo> listener) {
+        if (!context.peerIdentity().equals(USERNAME_1) && !context.peerIdentity().equals(USERNAME_2)) {
+          listener.onError(new IllegalArgumentException("Invalid username"));
+          return;
+        }
+        listener.onCompleted();
+      }
+
+      @Override
+      public void getStream(CallContext context, Ticket ticket, ServerStreamListener listener) {
+        if (!context.peerIdentity().equals(USERNAME_1) && !context.peerIdentity().equals(USERNAME_2)) {
+          listener.error(new IllegalArgumentException("Invalid username"));
+          return;
+        }
+        final Schema pojoSchema = new Schema(ImmutableList.of(Field.nullable("a",
+                Types.MinorType.BIGINT.getType())));
+        try (VectorSchemaRoot root = VectorSchemaRoot.create(pojoSchema, allocator)) {
+          listener.start(root);
+          root.allocateNew();
+          root.setRowCount(4095);
+          listener.putNext();
+          listener.completed();
+        }
+      }
+    };
+
+    serverWithBasicAuth = FlightTestUtil.getStartedServer((location) -> FlightServer
```

Comment:
Let's not do this in setup() and just have a helper method for setting up test state that each test can invoke (ie parameters can be the auth type, number of clients, etc).

What you're doing is allocating everything for every test then cleaning everything up for every test. It seems like alot of overhead.

Reply:
Updated tests to have better resource allocation.